Simplenote/SPTextView.swift

```diff
@@ -0,0 +1,128 @@
+import AppKit
+
+// MARK: - SPTextView
+//
+class SPTextView: NSTextView {
+
+    /// Is called when text view resigns being first responder
+    ///
+    var onResignFirstResponder: (() -> Void)?
+
+    /// Highlighted ranges
+    ///
+    var highlightedRanges: [NSRange] = [] {
+        didSet {
+            guard let textStorage = textStorage else {
+                return
+            }
+
+            for range in oldValue where range.upperBound <= textStorage.fullRange.upperBound {
+                textStorage.removeAttribute(.backgroundColor, range: range)
+            }
```

Comment:
I'm removing highlights by removing `backgroundColor` attributes from the text. From my quick research I saw that we're not using `backgroundColor` attribute for anything else so I decided to go with this approach as it is more efficient than just setting the text again.

Reply:
We're not using it, your approach is solid!!